mesonbuild/dependencies/boost.py

```diff
@@ -561,7 +557,7 @@ def detect_lib_dirs(self, root: Path) -> T.List[Path]:
         # Filter out paths that don't match the target arch to avoid finding
         # the wrong libraries. See https://github.com/mesonbuild/meson/issues/7110
         if not self.arch:
-            return dirs + subdirs
+            return system_dirs + dirs + subdirs
```

Comment:
`system_dirs` should also be returned here in an `elif` branch since they would be ignored otherwise and should not be subject to the filtering logic below

Reply:
After reading #7110 I think I might be going about this wrong let me try a different approach

Reply:
I re-wrote it to work explicitly what I was trying to do. Now when searching BOOST_ROOT it won't use the system_dirs, but when looking in other directories like /usr/local it will